lightbar_driver/src/lightbar_driver_application.cpp

```diff
@@ -56,42 +72,40 @@ void LightBarApplication::initialize()
     }
     catch (CURL_EASY_PERFORM_ERROR& e)
     {
-        ROS_WARN_STREAM(e.what());
-        ROS_WARN_STREAM("When starting lightbar driver, could not connect to the lightbar IP: " + host_name);
-        cav_msgs::DriverStatus status = getStatus();
-        status.status = cav_msgs::DriverStatus::FAULT;
+        RCLCPP_WARN_STREAM(get_logger(),e.what());
+        RCLCPP_WARN_STREAM(get_logger(),"When starting lightbar driver, could not connect to the lightbar IP: " + host_name);
+        carma_driver_msgs::msg::DriverStatus status = getStatus();
+        status.status = carma_driver_msgs::msg::DriverStatus::FAULT;
         setStatus(status);
         return;
     }
-
-    // Setup the ROS API
-    api_.clear();
-    
-    // Topics 
-    lightbar_pub_ = nh_->advertise<cav_msgs::LightBarStatus>("light_bar_status", 10);
-    api_.push_back(lightbar_pub_.getTopic());
-
-    // Services
-    get_lights_srv_ = nh_->advertiseService("get_lights", &LightBarApplication::getLightsCB, this);
-    api_.push_back(get_lights_srv_.getService());
-
-    set_lights_srv_ = nh_->advertiseService("set_lights", &LightBarApplication::setLightsCB, this);
-    api_.push_back(set_lights_srv_.getService());
-
-    // Timer loop
-    status_publisher_timer_ = nh_->createWallTimer(ros::WallDuration(ros::Rate(status_update_rate)),&LightBarApplication::updateStatusTimerCB, this);
-    status_publisher_timer_.start();
-
+   
     // Driver Status set to Operational
-    cav_msgs::DriverStatus status = getStatus();
-    status.status = cav_msgs::DriverStatus::OPERATIONAL;
+    carma_driver_msgs::msg::DriverStatus status = getStatus();
+    status.status = carma_driver_msgs::msg::DriverStatus::OPERATIONAL;
     setStatus(status);
+
+    // Return success if everthing initialized successfully
+    return CallbackReturn::SUCCESS;
+  } 
+
     
-}
+carma_ros2_utils::CallbackReturn LightBarApplication::handle_on_activate(const rclcpp_lifecycle::State &prev_state)
+ {
+        //Timer setup - Light Bar
+        lightbar_timer_ = this->create_wall_timer(std::chrono::milliseconds(500), 
+        std::bind(&LightBarApplication::updateStatusTimerCB, this));
 
-bool LightBarApplication::getLightsCB(cav_srvs::GetLightsRequest&, cav_srvs::GetLightsResponse &resp) {
-    static auto ON = static_cast<cav_msgs::LightBarStatus::_flash_type>(cav_msgs::LightBarStatus::ON);
-    static auto OFF = static_cast<cav_msgs::LightBarStatus::_flash_type>(cav_msgs::LightBarStatus::OFF);
+        //Timer setup - Driver Discovery
+        driver_discovery_timer_ = this->create_wall_timer(std::chrono::milliseconds(500), 
```

Comment:
Let's make it 1000.
it used to be 1 sec here:
https://github.com/usdot-fhwa-stol/carma-utils/blob/develop/cav_driver_utils/src/driver_application/driver_application.cpp#L58